src/zip.h

```diff
@@ -373,6 +380,24 @@ extern ZIP_EXPORT ssize_t zip_entry_read(struct zip_t *zip, void **buf,
 extern ZIP_EXPORT ssize_t zip_entry_noallocread(struct zip_t *zip, void *buf,
                                                 size_t bufsize);
 
+/**
+ * Extracts the part of the current zip entry into a memory buffer using no memory
+ * allocation for the buffer.
+ *
+ * @param zip zip archive handler.
+ * @param offset the offset of the entry (in bytes).
+ * @param size requested number of bytes (in bytes).
+ * @param buf preallocated output buffer.
+ *
+ * @note the iterator api uses an allocation to create its state
+ * @note each call will iterate from the start of the entry
+ *
+ * @return the return code - the number of bytes actually read on success.
+ *         Otherwise a negative number (< 0) on error (e.g. offset is too large).
+ */
+extern ZIP_EXPORT ssize_t zip_entry_noallocread_offset(struct zip_t *zip,
```

Comment:
Maybe to follow similar API, we can name it `zip_entry_noallocreadwithoffset`. I know it's not the prettiest name, but at least it's consistent with other API `...witherror`, `...withindex`, etc.